src/components/profile/privacy-and-security/password-modal.vue

```diff
@@ -0,0 +1,153 @@
+                                                :inputValue="newPass"
+                                                placeholder="Type in your new password"
+                                                @input="newPass = $event"
+                                            />
+                                            <InputPassword
+                                                id="confirm-password"
+                                                label="Confirm password"
+                                                type="text"
+                                                :inputValue="confirmPass"
+                                                placeholder="Confirm your new password"
+                                                @input="confirmPass = $event"
+                                            />
+                                        </div>
+                                    </div>
+                                </form>
+                            </div>
+                        </div>
+                        <button type="button" class="btn btn-primary btn-submit w-100 m-0 mt-auto text-light" v-show="step === 1" @click="sendVerificationCode">
+                            Send verification code
+                        </button>
+                        <button type="button" class="btn btn-primary btn-submit w-100 m-0 mt-auto text-light" v-show="step === 2" @click="submit">
+                            Save new password
+                        </button>
+                    </div>
+                </div>
+            </Teleport>
+        </client-only>
+    </div>
+</template>
+
+<script setup>
+import { PhPencil, PhUserCircle, PhX } from "@phosphor-icons/vue";
+import InputText from '@/components/input-fields/input-text.vue';
+import InputPassword from "@/components/input-fields/input-password.vue";
+const verificationCode = ref('');
+const newPass = ref('');
+const confirmPass = ref('');
+let step = ref(1);
+const sendVerificationCode = () => {
+    step.value = 2;
+};
+const open = ref(false);
+const showModal = () => {
+    open.value = true;
+    resetData();
+};
+const closeModal = () => {
+    open.value = false;
+};
+const submit = () => {
+    if (!verificationCode.value || !newPass.value || !confirmPass.value) {
+        console.log('all fields must be filled');
+        return;
+    }
+    if (newPass.value !== confirmPass.value) {
+        console.log('passwords do not match');
+        return;
+    }
+    changePassword();
+    closeModal();
+};
+const changePassword = () => {
+    console.log('change password');
+};
+const resetData = () => {
+    verificationCode.value = '';
+    newPass.value = '';
+    confirmPass.value = '';
+    step.value = 1;
+};
+</script>
+
+<style scoped lang="scss">
+.stepper__container {
+    height: -webkit-fill-available!important;
+}
+.modal-box {
+    display: none;
+    position: fixed;
+    z-index: 999;
```

Comment:
z-index is usually 0, 100, 1000 etc.

Reply:
@Mozes721 This review is not for this PR, rather #10. Kindly make your review there. The reason for this has already been mentioned in the description when the PR was initiated. Resolving this conversation.